### PR TITLE
New modules for landing page

### LINKED ISSUE #20: Default template links are orange on hover
Links on hover/active are orange (default template).

**Expected behavior**
Links on hover/active should be blue (#0873e8)

### LINKED ISSUE #23: navbar not displayed inline on mobile version
**Describe the bug**
when screen size is small (phone size) the navbar links are not displayed inline.
**To Reproduce**
1.reduce screen size to smallest or check mobile version on inspector.

**Expected behavior**
navbar links should be displayed inline on every screen size.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by PolartJacques**
## Related Issues

Fixes #20

### CODE REVIEW

File: src/components/Landing/skills/skills.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <div className={styles['text']}>
+      <h3>Notre expertise</h3>
+      <p>pour rester à la pointe de la technologie</p>
+    </div>
+    <GatsbyImage image={logos[0].node.childImageSharp.gatsbyImageData} alt="aws" className={styles['logo']} style={{ top: '20%', left: '30%' }}/>
+    <GatsbyImage image={logos[1].node.childImageSharp.gatsbyImageData} alt="azure" className={styles['logo']} style={{ top: '20%', left: '50%' }}/>
+    <GatsbyImage image={logos[2].node.childImageSharp.gatsbyImageData} alt="cloud" className={styles['logo']} style={{ top: '20%', left: '70%' }}/>
+    <GatsbyImage image={logos[3].node.childImageSharp.gatsbyImageData} alt="docker" className={styles['logo']} style={{ top: '80%', left: '30%' }}/>
+    <GatsbyImage image={logos[4].node.childImageSharp.gatsbyImageData} alt="kubernetes" className={styles['logo']} style={{ top: '80%', left: '50%' }}/>
+    <GatsbyImage image={logos[5].node.childImageSharp.gatsbyImageData} alt="prometheus" className={styles['logo']} style={{ top: '80%', left: '70%' }}/>
```
Comment (AlexHff) [inline src/components/Landing/skills/skills.js]: Always be careful when referencing an array item. What if it's empty by mistake? The whole page might crash. You should check that each image was retrieved successfully.

File: gatsby/create-pages.js [hunk 2]
```diff
@@ -12,27 +12,85 @@ const createPages = async ({ graphql, actions }) => {
...
+              slug
+            }
+            frontmatter {
+              date
+              description
+              title
+              category
+              socialImage {
+                childImageSharp {
+                  gatsbyImageData
```
Comment (AlexHff) [inline gatsby/create-pages.js]: Why did you use this piece of data?

File: gatsby/create-pages.js [hunk 3]
```diff
@@ -12,27 +12,85 @@ const createPages = async ({ graphql, actions }) => {
...
+    }
+  `);
+
+  const skills = await graphql(`
+    {
+      allFile(filter: {relativeDirectory: {eq: "skills"}}) {
+        edges {
+          node {
+            childImageSharp {
+              gatsbyImageData(width: 64, height: 64)
```
Comment (AlexHff) [inline gatsby/create-pages.js]: You should save the names of the skills in the data and display the entire section as an array using a `map()` function.

File: gatsby/create-pages.js [hunk 4]
```diff
@@ -12,27 +12,85 @@ const createPages = async ({ graphql, actions }) => {
...
+    }
+  `);
+
+  const clients = await graphql(`
+    {
+      allFile(filter: {relativeDirectory: {eq: "clients"}}) {
+        edges {
+          node {
+            childImageSharp {
+              gatsbyImageData(height: 80)
```
Comment (AlexHff) [inline gatsby/create-pages.js]: Also save the names of the clients in data for the alt attribute (and more).

File: src/components/Landing/Clients/Clients.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { GatsbyImage } from 'gatsby-plugin-image';
+import React from 'react';
+import styles from './Clients.module.scss';
+
+const Clients = ({ clients }) => (
+  <div className={styles['container']}>
+    <h1>Ils nous font confiance</h1>
+    <div className={styles['logos']}>
+      {clients.map((client, i) => (
+        <GatsbyImage className={styles['logo']} image={client.node.childImageSharp.gatsbyImageData} alt="client" key={i} imgStyle={{ objectFit: 'contain' }} />
```
Comment (AlexHff) [inline src/components/Landing/Clients/Clients.js]: Good initiative to use a map here. As mentioned in a comment above, `alt` attribute should be saved in client data and printed here. The `key` attribute should also be associated with the customer name (it's more relevant than the index). You are missing the `width` and `height` attributes.

File: src/components/Landing/Landing.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            gravida sed eget nulla. Mauris id tortor at mi varius lobortis nec id
+            augue. Curabitur efficitur augue quis diam convallis, vel dictum
+            tortor pulvinar.
+          </p>
+        </div>
+      </Tabs>
+    </div>
+    <Clients clients={clients} />
+    <Skills skills={skills} />
+    <h1>Nos derniers posts</h1>
```
Comment (AlexHff) [inline src/components/Landing/Landing.js]: This should be in the `LastPosts` component if it's not used somewhere else. Even then, seems more relevant.

File: src/components/Landing/LastPosts/LastPosts.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Link, navigate } from 'gatsby';
+import { GatsbyImage } from 'gatsby-plugin-image';
+import styles from './LastPosts.module.scss';
+
+const LastPosts = ({ lastPosts }) => (
+    <div className={styles['container']}>
+      {lastPosts.map((post, i) => (
+        <div key={i} className={styles['card']}>
```
Comment (AlexHff) [inline src/components/Landing/LastPosts/LastPosts.js]: You can use the slug as a key. It's more relevant than the index.

File: src/components/Landing/LastPosts/LastPosts.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+          <p className={styles['category']}>{post.node.frontmatter.category.toUpperCase()}</p>
+          <GatsbyImage
+            image={post.node.frontmatter.socialImage.childImageSharp.gatsbyImageData}
+            alt={post.node.frontmatter.title}
+            onClick={() => navigate(post.node.fields.slug)}/>
```
Comment (AlexHff) [inline src/components/Landing/LastPosts/LastPosts.js]: You're using the `navigate()` function to go to the post. Is there a reaso why you're not using `Link`? How to `navigate()` work?

File: src/components/Landing/LastPosts/LastPosts.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import { Link, navigate } from 'gatsby';
+import { GatsbyImage } from 'gatsby-plugin-image';
+import styles from './LastPosts.module.scss';
+
+const LastPosts = ({ lastPosts }) => (
+    <div className={styles['container']}>
+      {lastPosts.map((post, i) => (
+        <div key={i} className={styles['card']}>
+          <p className={styles['category']}>{post.node.frontmatter.category.toUpperCase()}</p>
+          <GatsbyImage
```
Comment (AlexHff) [inline src/components/Landing/LastPosts/LastPosts.js:16]: You're missing `width` and `height` attributes.

File: src/components/Landing/LastPosts/LastPosts.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+            image={post.node.frontmatter.socialImage.childImageSharp.gatsbyImageData}
+            alt={post.node.frontmatter.title}
+            onClick={() => navigate(post.node.fields.slug)}/>
+          <h4 onClick={() => navigate(post.node.fields.slug)}>{post.node.frontmatter.title}</h4>
```
Comment (AlexHff) [inline src/components/Landing/LastPosts/LastPosts.js]: Same questions about `navigate()` than above.

File: src/components/Landing/LastPosts/LastPosts.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+          <p>{post.node.frontmatter.description}</p>
+          <Link to={post.node.fields.slug} className={styles['button']}>voir l'article</Link>
```
Comment (AlexHff) [inline src/components/Landing/LastPosts/LastPosts.js]: Capitalize the first letter of the button.

File: src/components/Landing/WelcomeModule/WelcomeModule.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import styles from './WelcomeModule.module.scss';
+
+const WelcomeModule = () => (
+  <div className={styles['welcome__module']}>
+    <img
+      className={styles['welcome__module__image']}
+      src='/workspace.png'
+    ></img>
```
Comment (AlexHff) [inline src/components/Landing/WelcomeModule/WelcomeModule.js]: Upgrade the image implementation using Gatsby Image. Don't forget to add the following attributes: `alt`, `width` and `height`.

File: src/components/Landing/WelcomeModule/WelcomeModule.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import styles from './WelcomeModule.module.scss';
+
+const WelcomeModule = () => (
```
Comment (AlexHff) [inline src/components/Landing/WelcomeModule/WelcomeModule.js]: This should be called `Hero`. Look up what a Hero is.

File: src/components/Landing/skills/skills.js [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+      <h3>Notre expertise</h3>
+      <p>pour rester à la pointe de la technologie</p>
+    </div>
+    {displaySkills(skills)}
+  </div>
+);
+
+const displaySkills = (skills: any[]) => {
+  const render = [];
+  skills.forEach((skill, i) => {
```
Comment (AlexHff) [inline src/components/Landing/skills/skills.js]: Why are you using `array.forEach()` in this component when you used `array.map()` in the other one?

File: src/components/Landing/skills/skills.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+    const angle = (i * (360 / skills.length)) * (Math.PI / 180);
+    render.push(<GatsbyImage image={skill.node.childImageSharp.gatsbyImageData} alt="aws" className={styles['logo']}
```
Comment (AlexHff) [inline src/components/Landing/skills/skills.js]: As mentioned in another comment, the use of a mapping function is good. Nonetheless, you should have a dynamiac `alt` attribute by saving it as static data. Also, add the `width` and `height` attributes.

File: src/components/Landing/skills/skills.js [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+      style={{ top: `${50 + Math.cos(angle) * 40}%`, left: `${50 + Math.sin(angle) * 40}%` }} />);
+  });
+  return render;
```
Comment (AlexHff) [inline src/components/Landing/skills/skills.js]: This implementation is interesting. We should stay consistent. In all other files, we render an entire `div` tag whereas here, you render an array that contains multiplie tags. Refactor the code to be consistent with the rest of the codebase.

File: src/components/Landing/skills/skills.js [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+    {displaySkills(skills)}
+  </div>
+);
+
+const displaySkills = (skills: any[]) => {
+  const render = [];
+  skills.forEach((skill, i) => {
+    const angle = (i * (360 / skills.length)) * (Math.PI / 180);
+    render.push(<GatsbyImage image={skill.node.childImageSharp.gatsbyImageData} alt="aws" className={styles['logo']}
+      style={{ top: `${50 + Math.cos(angle) * 40}%`, left: `${50 + Math.sin(angle) * 40}%` }} />);
```
Comment (AlexHff) [inline src/components/Landing/skills/skills.js]: Inline style is discouraged.

File: src/components/Landing/skills/skills.js (hunk 16 above)
Comment (AlexHff) [reply src/components/Landing/skills/skills.js]: We can discuss this further.

File: src/templates/landing-template.js [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+        }
+      }
+    }
+  }
+}
+
+const LandingTemplate = ({ pageContext }: Props) => {
+  const lastPosts = pageContext.lastPosts.data.allMarkdownRemark.edges;
+  const clients = pageContext.clients.data.allFile.edges;
+  const skills = pageContext.skills.data.allFile.edges;
```
Comment (AlexHff) [inline src/templates/landing-template.js]: What is these values are `null` or `undefined`. Here is the place to check that everything worked as expected before passing the data to other Components.
ALWAYS check for limit cases (what if something is null or infinite).

**Review by AlexHff — CHANGES_REQUESTED** [review]
Thanks for the initial work @EliottElek @PolartJacques. There are some elements that need change. Try and improve Lighthouse score **as a final step**.

File: gatsby/create-pages.js (hunk 2 above)
Comment (PolartJacques) [reply gatsby/create-pages.js]: i just folowed the documentaion
we don't use gatsby-image anymore as it's deprecated, bu gatsby-plugin-image

File: src/components/Landing/Clients/Clients.js (hunk 5 above)
Comment (PolartJacques) [reply src/components/Landing/Clients/Clients.js]: are width and height attributes really usefull ? by default the GatsbyImage tag create an attribute size depending on the gatsbyImageData
exemple : sizes="(min-width: 64px) 64px, 100vw"

Comment (AlexHff) [reply src/components/Landing/Clients/Clients.js]: This is mandatory for Google Lighthouse tests and SEO.

File: src/components/Landing/skills/skills.js (hunk 17 above)
Comment (PolartJacques) [reply src/components/Landing/skills/skills.js]: top and left attribute are calculated for each image, i did'nt found a working way to implement this directly into the scss file

File: src/components/Landing/Tabs/Tabs.js [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+    e.preventDefault();
+    setActiveTab(newActiveTab);
+  };
+
+  return (
+    <div className={styles['tabs__module']}>
+      <ul className={styles['tabs__list']}>
+        {/* Fonction map à vérifier */}
+        {children.map((tab) => (
+          <li
```
Comment (AlexHff) [inline src/components/Landing/Tabs/Tabs.js]: Remove the entire file.

Comment (AlexHff) [reply src/components/Landing/Tabs/Tabs.js]: The entire component

**Review by AlexHff — CHANGES_REQUESTED** [review]

**Comment by AlexHff** [comment]
Good job, try and rebase `master`.

File: gatsby/on-render-body.js [hunk 20]
```diff
@@ -21,6 +21,7 @@ const onRenderBody = ({ setPostBodyComponents }) => {
       `,
       }}
     />,
+    <script type="text/javascript" id="hs-script-loader" async defer src="//js-na1.hs-scripts.com/20445448.js"></script>
```
Comment (AlexHff) [inline gatsby/on-render-body.js]: This shouldn't be here.

File: package.json [hunk 21]
```diff
@@ -45,9 +45,11 @@
     "gatsby-plugin-feed": "^2.13.1",
     "gatsby-plugin-flow": "^1.10.0",
     "gatsby-plugin-google-gtag": "^2.8.0",
+    "gatsby-plugin-image": "^1.10.0",
     "gatsby-plugin-mailchimp": "^5.2.2",
     "gatsby-plugin-manifest": "^2.12.1",
     "gatsby-plugin-netlify": "^2.11.1",
+    "gatsby-plugin-netlify-cms": "^5.10.0",
```
Comment (AlexHff) [inline package.json]: This was removed on master

File: src/components/Footer/MiniLogo.js [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+const MiniLogo = ({ img }: Props) => (
+  <div className={styles['logo']}>
+    <Link to="/">
+      <img
+        src={withPrefix(img)}
+        className={styles['logo__photo']}
+        width="30"
+        height="30"
+        alt="logo"
+      />
```
Comment (AlexHff) [inline src/components/Footer/MiniLogo.js]: Use Gatsby image to implement the img

File: src/components/Landing/Services/Services.js [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+            />
+            <h4>DevOps & InfraOps</h4>
+            <p className={styles['services__module__innergrid__comp__text']}>
+              Ansible, Docker, Git, Kubernetes, Maven, Unix, Vagrant
+            </p>
+          </>
+        )}
+      </div>
+      <div className={styles['services__module__innergrid__comp']}>
+        {services[2] && (
```
Comment (AlexHff) [inline src/components/Landing/Services/Services.js]: What about services[1]?

**Review by AlexHff — CHANGES_REQUESTED** [review]

**Comment by AlexHff** [comment]
All issues that were resolved in the last commit:
- Implemented types for all Components;
- Added unresolved modules to flow types;
- Fixed `Navbar` unit test;
- Fixed generic formatting issues.

Do NOT forget to implement flow checking for each new Component.
Thanks.

File: src/components/Landing/Landing.js [hunk 24]
```diff
@@ -0,0 +1,25 @@
...
+import CallToAction from '../CallToAction';
+
+const Landing = ({
+  lastPosts, clients, skills, landing, services
+}) => (
+  <div className={styles['landing__page']}>
+    <Hero landing={landing} />
+    <CallToAction />
+    <Clients clients={clients} />
+    <div className={styles['services']}>
```
Comment (AlexHff) [inline src/components/Landing/Landing.js]: This line should be in the Services component

File: src/components/Landing/Landing.js [hunk 25]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
Comment (AlexHff) [inline src/components/Landing/Landing.js:2]: Why is there no flow check?

File: src/components/Landing/Services/Services.js [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+      name: string,
+      description: string,
+      logo: {
+        childImageSharp: {
+          gatsbyImageData: typeof gatsbyImageData
+        }
+      }
+    }
+  }
+};
```
Comment (AlexHff) [inline src/components/Landing/Services/Services.js]: Add one line break

File: src/components/Landing/Services/Services.js [hunk 27]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
```
Comment (AlexHff) [inline src/components/Landing/Services/Services.js:2]: Add flow check

File: src/hooks/use-global-data.js [hunk 28]
```diff
@@ -0,0 +1,27 @@
...
+              }
+              name
+            }
+          }
+        }
+      }
+    `
+  );
+
+  const logo = data.allFile?.edges[0]?.node ? data.allFile?.edges[0]?.node : undefined;
```
Comment (AlexHff) [inline src/hooks/use-global-data.js:22]: Maybe throw an error or a warning. It seems like `logo` would be undefined by default so what's this check for?

Comment (PolartJacques) [reply src/hooks/use-global-data.js:22]: if logo not found, fields allFile and node are not present in the graphql response, and it make the site crash

Comment (AlexHff) [reply src/hooks/use-global-data.js:22]: Ok thanks

File: src/components/Landing/EndingCallToAction/EndingCallToAction.js [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+type Props = {
+  node: {
+    childImageSharp: typeof gatsbyImageData,
+    name?: string
+  }
+};
+
+const EndingCallToAction = ({ landing }: { landing: Array<Props> }) => (
+  <div className={styles['container']}>
+    {landing[1] && (
```
Comment (AlexHff) [inline src/components/Landing/EndingCallToAction/EndingCallToAction.js]: Classical return

File: src/components/Landing/HeroModule/HeroModule.js [hunk 30]
```diff
@@ -0,0 +1,35 @@
...
+type Props = {
+  node: {
+    childImageSharp: typeof gatsbyImageData,
+    name?: string
+  }
+};
+
+const Hero = ({ landing }: { landing: Array<Props> }) => (
+  <div className={styles['hero__module']}>
+    {landing[0] && (
```
Comment (AlexHff) [inline src/components/Landing/HeroModule/HeroModule.js]: Classic return

File: src/components/Landing/Landing.js [hunk 31]
```diff
@@ -0,0 +1,38 @@
...
+  <div className={styles['landing__page']}>
+    <Hero landing={landing} />
+    <CallToAction />
+    <Clients clients={clients} />
+    <Services services={services} />
+    <Stats />
+    <Skills skills={skills} />
+    <LastPosts lastPosts={lastPosts} />
+    <EndingCallToAction landing={landing} />
+
```
Comment (AlexHff) [inline src/components/Landing/Landing.js]: remove empty line

File: src/types/index.js [hunk 32]
```diff
@@ -58,4 +59,25 @@ export type AllMarkdownRemark = {
...
+  node: {
+    frontmatter: {
+      description: string,
+      name: string,
+      logo: {
+        childImageSharp: typeof gatsbyImageData;
+      }
+    }
+  }
+}
```
Comment (AlexHff) [inline src/types/index.js:81]: Move to Service

File: src/types/index.js [hunk 33]
```diff
@@ -58,4 +59,25 @@ export type AllMarkdownRemark = {
   }[]
 };
 
+export type GatsbyImage = {
+  node: {
+    childImageSharp: typeof gatsbyImageData;
+    name?: string;
+  }
+}
```
Comment (AlexHff) [inline src/types/index.js]: Use in all components that implement a Gatsby image.

**Review by AlexHff — CHANGES_REQUESTED** [review]

File: src/types/index.js (hunk 32 above)
Comment (PolartJacques) [reply src/types/index.js:81]: ServiceData type is used in other files than services.js, so it has to be exported anyway.
Should it be exported from services.js ? or can we leave it there

Comment (AlexHff) [reply src/types/index.js:81]: Leave it there, thanks.

File: .env.production [hunk 34]
```diff
@@ -0,0 +1 @@
+GATSBY_WEBHOOK_URL=https://hooks.zapier.com/hooks/catch/10583025/b28wqv3/silent/
```
Comment (AlexHff) [inline .env.production]: Remove

File: .env.development [hunk 35]
```diff
@@ -0,0 +1 @@
+GATSBY_WEBHOOK_URL=https://hooks.zapier.com/hooks/catch/10583025/b28wqv3/silent/
```
Comment (AlexHff) [inline .env.development]: Remove

File: yarn-error.log [hunk 36]
```diff
@@ -0,0 +1,18728 @@
...
+    dependencies:
+      delayed-stream "~1.0.0"
+  
+  comma-separated-tokens@^1.0.0, comma-separated-tokens@^1.0.1:
+    version "1.0.8"
+    resolved "https://registry.yarnpkg.com/comma-separated-tokens/-/comma-separated-tokens-1.0.8.tgz#632b80b6117867a158f1080ad498b2fbe7e3f5ea"
+    integrity sha512-GHuDRO12Sypu2cV70d1dkA2EUmXHgntrzbpvOB+Qy+49ypNfGgFQIC2fhhXbnyrJRynDCAARsT7Ou0M6hirpfw==
+  
+  command-exists@^1.2.4:
+    versi
```
Comment (AlexHff) [inline yarn-error.log]: Remove

**Review by AlexHff — CHANGES_REQUESTED** [review]

**Comment by AlexHff** [comment]
Good work @PolartJacques @EliottElek. Thanks!

**Pull request merged**